mockld/polling_service.go

```diff
@@ -38,23 +40,26 @@ const (
 )
 
 type PollingService struct {
-	sdkKind     SDKKind
-	currentData SDKData
-	currentEtag string
-	handler     http.Handler
-	debugLogger framework.Logger
-	lock        sync.RWMutex
+	sdkKind               SDKKind
+	currentData           SDKData
+	currentEtag           string
+	handler               http.Handler
+	enableGzipCompression bool
+	debugLogger           framework.Logger
+	lock                  sync.RWMutex
 }
 
 func NewPollingService(
 	initialData SDKData,
 	sdkKind SDKKind,
+	enableGzipCompression bool,
 	debugLogger framework.Logger,
```

Comment:
I think it could be a good time to refactor this to be more go-ish, with optional arguments via adding a `WithGzipCompression(bool)` method to `*PollingService`. So:
```golang
p := NewPollingService(x, y, z).WithGzipCompression(true)
```